jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -178,6 +190,70 @@ public BuildConfiguration build() {
           throw new IllegalStateException(errorMessage.toString());
       }
     }
+
+    /**
+     * Converts/validates a list of ports with ranges to an expanded form without ranges.
+     *
+     * <p>Example: ["1000/tcp", "2000-2002/tcp"] -> ["1000/tcp", "2000/tcp", "2001/tcp", "2002/tcp"]
+     *
+     * @param ports the list of port numbers/ranges
+     * @return the ports as a list of integers
+     * @throws NumberFormatException if any of the ports are in an invalid format or out of range
+     */
+    @VisibleForTesting
+    ImmutableList<String> expandPortRanges(List<String> ports) throws NumberFormatException {
```

Comment:
Let's add a TODO for moving this to a class in frontend, so that BuildConfiguration can be kept free of user input parsing.